Add Logging Feature

dwieyoko opened the pull request.

contactjavas commented:
Hi @dwieyoko , thanks for the good PR.
Below are some points that I think need attention:

1. **Question**
The `capture_email_details_for_logging` method is hooked into `wp_mail` filter in `class-logs-module.php`. I see this function doesn't do anything for the arguments. Meaning, it's better to use action hook (if there's any) instead of filter hook. Is there an action hook that we can use instead of filter hook?
2. **Function's parameter type**
Let's use correct type for function's param. For instance, the `handle_failed_email` method in `class-logs-module.php` has `$wp_error` parameter. The function's comment (the `@param` part) doesn't have correct type.
3. **Question**
If we look at `determine_email_type` method in `class-logs-module.php`, the email type is based on part of the subject's string. Is this reliable enough?
4. **Consideration**
In `get_sender_email` method in `class-logs-module.php`, we get the sender email from our settings. Can we get the real sender value of by checking the current processed email instead of checking to our setting?
5. **Suggestion**
Please prefix the post type with `weed` to prevent conflict possiblity with other plugin/3rd parties. Currently it's `email_logs` which is quite general.
6. **Suggestion**
About capability, I think it's better to set the allowed capability on `register_post_type` and `add_submenu_page` instead of manually hooking to current_screen like in `class-logs-output.php` file.

And for coding style, I added `phpcs.xml` to this repo. Since your vscode seems already setup, you might be able to auto-format it using `Shift Alt F` keys shortcut.

dwieyoko commented:
Hi @contactjavas , thanks for the nice feedbacks.
Below are the answers of them:
1. Yes, there is a WP filter that we can use to replace them. It's `pre_wp_mail`  filter that will fire before the email being sent. With this filter, we can intercept email data for our needs.
2. Solved.
3. No, but currently I have deleted that function, because David request to change it with the email subject.
4. Yes, we can use `phpmailer_init` action to capture the email sender.
5. Alright, changed to `weed_email_logs`.
6. Alright, I have set the capability to register post type and when adding sub menu item
